src/components/pagination/pagination.js

```diff
@@ -95,6 +103,25 @@ export default class Pagination extends React.PureComponent {
         {delta > 0 && (
           <>
             <ul className="pagination-list">
+              {showFirstLast && current !== 1 && firstPage !== 1 && (
+                <>
+                  <li key={1}>
+                    <a
+                      role="button"
+                      tabIndex={0}
+                      className="pagination-link"
+                      onClick={current === 1 ? undefined : this.goToPage(1)}
```

Comment:
I think we can remove this `current === 1` . you already check this condition on line 106

Reply:
Thanks for pointing that out. i have made another commit to remove the duplicated checking on both first and last buttons.